sirepo/sim_data/flash.py

```diff
@@ -275,12 +275,11 @@ def __put_sim_files(cls, data, datafiles, make_dir, run_dir):
             .items()
         ):
             p = make_dir.join(c)
-            # TODO(pjm): this call broke the LaserSlab example
-            #            files were possibly being deleted after the put occurred
-            # cls.delete_sim_file(
-            #     data.models.simulation.simulationId,
-            #     b.split(cls._FLASH_FILE_NAME_SEP)[0],
-            # )
+            if b.startswith(cls._FLASH_EXE_PREFIX):
```

Comment:
What about moving this outside of the loop and making it a named function (maybe even with a docstring)? It is a tricky case that we are trying to delete the exe with a glob that is added for us on the receiving end. I think that deserves a special callout. 


```
def _delete_flash_exe():
   """Delete the flash executable if it exists
    
    The executables are named like _FLASH_EXE_PREFIX_<unique-hash>.
    They have a hash so we know when we need to recreate the executable
    based on the user changing parameters.
    
   sirepo.sim_db_file delete does a glob of the filename to delete. So, we
   will delete any file that starts with _FLASH_EXE_PREFIX
   """"
   cls.delete_sim_file(data.models.simulation.simulationId, cls._FLASH_EXE_PREFIX)

Reply:
Yes, sounds good - I made this change with 3ddf958